developer_manual/scripts/theme-bootstrap.sh

```diff
@@ -45,42 +61,72 @@ function clone_example_theme
 
 E_BADARGS=85
 
-# Stores the directory where this script was called from
+# Remembers the directory where this script was called from
 SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
 
+# Check if run as sudo (root), needed for sub script calling and changing file permissions
+if (( $EUID != 0 )); then
+    echo "Please run this script with sudo or as root"
+    exit
+fi
+
+# Check if enough parameters have been applied
 if (( $# != 2 ))
 then
-  echo "Not enough arguments supplied."
-  echo "Usage: $( basename "$0" ) [-v] [-h] [theme id] [owncloud directory]"
+  echo "Not enough arguments provided."
+  echo "Usage: $( basename "$0" ) [new theme name] [owncloud root directory]"
   exit $E_BADARGS
 fi
 
+# Check if read-config.php file exists in the same directory
+if [ ! -f $SCRIPT_DIR/read-config.php ]
+then
+    echo "File read-config.php not found! Must be in the same dir as this script"
+    exit
+fi
+
+# Check if php file is set to be executable, script will else not work
+if [ ! -x $SCRIPT_DIR/read-config.php ]
+then
+    echo "File read-config.php is not set executable"
+    exit
+fi
+
 app_name="$1"
-owncloud_directory="$2"
-apps=$("$SCRIPT_DIR/read-config.php" "$owncloud_directory")
+owncloud_root="$2"
+apps=$(php "$SCRIPT_DIR/read-config.php" "$owncloud_root")
 
-OLDPWD=${OLDPWD:=$(pwd)}
+# Check if the php script returned an error message. This is when the straing does not start with /
```

Comment:
What's a "straing"? heh :-)

Reply:
a straing is part of the string theory and applied instead a string when your brain vibrates to much 😄